File: src/engine/commands/quitcommand.cpp
#include "quitcommand.h"

#include "util.h"


#define super Command

QuitCommand::QuitCommand(QObject *parent) :
    super(parent) {

    setDescription("Quit the game. Your progress and current location will be saved.\n"
                   "\n"
                   "Examples: quit, goodbye");
}

QuitCommand::~QuitCommand() {
}

void QuitCommand::execute(Player *player, const QString &command) {

    super::execute(player, command);

    player->quit();
}

File: data/commands/quitcommand.js
function QuitCommand() {
    Command.call(this);

    this.setDescription("Quit the game. Your progress and current location will be saved.\n" +
                        "\n" +
                        "Examples: quit, goodbye");
}

QuitCommand.prototype = new Command();
QuitCommand.prototype.constructor = QuitCommand;

QuitCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    player.quit();
}

CommandRegistry.registerCommand("goodbye", new QuitCommand());
CommandRegistry.registerCommand("quit", new QuitCommand());
